### ISSUE TITLE
FilePart<File> is null in binded Form

### ISSUE THREAD

**Issue by eximius313** [playframework/playframework#8323]
### Play Version (2.5.x / etc)
2.6.11

### API (Scala / Java / Neither / Both)
Java

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)
Windows 7

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)
JDK 9.0.4

### Library Dependencies
N/A

### Expected Behavior
`FilePart<File>` in binded form contains the file

### Actual Behavior
`FilePart<File>` in binded form is null.
[Documentation](https://www.playframework.com/documentation/2.6.x/JavaFileUpload#Uploading-files-in-a-form-using-multipart/form-data) shows only this example `request().body().asMultipartFormData()` and it works, but more often you have also other fields in form. Unfortunatelly `FilePart<File>`  used in form along with other fields is always null;

### Reproducible Test Case
```
public class MyForm {
      @Required
      private String email;

      private FilePart<File> photo;

      //setters & getters
}
```
```
public Result update() {
      final Form<MyForm> form = formFactory.form(MyForm.class).bindFromRequest();
      if (form.hasErrors()) {
            return badRequest(form.errorsAsJson());
      }
      final FilePart<File> photo = form.getPhoto(); <-- this is null

      final MultipartFormData<File> data = request().body().asMultipartFormData();
      final FilePart<File> photoDirect = data.getFile("photo"); <-- this is not null

      return ok();
}

```

**Comment by mkurz** [comment]
Hi @eximius313,

you don't have to define `photo` in `MyForm`:

```java
public class MyForm {
      @Required
      private String email;

      // private FilePart<File> photo; -> this is not needed in MyForm

      //setters & getters
}
```

**Comment by eximius313** [comment]
But i **want** to define it there and want it working ;)
The reason for this issue is to be able to use FilePart<File> photo as a regular element in the form.
Imagine, that you have "UserProfileForm" and with one "POST" user is sending name, email, birthdate and profile picture.
I want to use them like that:
```
public Result update() {
      final Form<UserProfileForm> form = formFactory.form(UserProfileForm.class).bindFromRequest();
      if (form.hasErrors()) {
            return badRequest(form.errorsAsJson());
      }
      final String email = form.getEmail(); 
      final String name = form.getName(); 
      final FilePart<File> photo = form.getPhoto(); 
      repository.save(User.of(email, name, photo))
      return ok();
}
```

**Comment by eximius313** [comment]
By the way - I remember it worked that way in Play 2.4

**Comment by mkurz** [comment]
@eximius313 
> By the way - I remember it worked that way in Play 2.4

Are you 100% sure? I was looking through the code and I don't think this ever worked...

**Comment by eximius313** [comment]
yes, I remember the pain of migration and how surprised I was when it stopped working

**Comment by mkurz** [comment]
Are you sure it was Play 2.4? Do you still have access to this project?

**Comment by eximius313** [comment]
yes, I have access.
I checked it, and now I know why it was working that way:
```
public class ProfileForm {

        @Constraints.Required
        private String email;

        private FilePart img;

        public List<ValidationError> validate() {
                MultipartFormData data = Application.request().body().asMultipartFormData();
                img = data.getFile("img");
                return null;
        }
}
```
so it was a hack :/

**Comment by mkurz** [comment]
Thanks for having a look!
As I thought, this wasn't and isn't currently supported by Play out of the box... I might add this feature...

**Comment by mkurz** [comment]
Hmm.. Actually that could be a nice feature.
E.g. when uploading an image in a form we could have constraints like:
```java
public class MyForm {
    @Required
    @MinFileSize(512)
    @MaxFileSize(4096)
    @AllowedExtensions({"jpg", "png"})
    @MaxDimensions(width = 1024, height = 768)
    private FilePart<File> photo;

    //setters & getters
}
```

**Comment by eximius313** [comment]
that would be AWESOME!

**Comment by eximius313** [comment]
I can't wait to use it! Finally normal validation for files! Thanks @mkurz !

**Comment by mkurz** [comment]
Fixed by #8898. Will be available in Play 2.7
